ctapipe/image/muon/processor.py

```diff
@@ -0,0 +1,211 @@
+from .ring_fitter import MuonRingFitter
+
+
+class DL1Query(QualityQuery):
+    quality_criteria = List(
+        Tuple(Unicode(), Unicode()),
+        default_value=[
+            ("min_pixels", "p.morphology.n_pixels > 100"),
+            ("min_intensity", "p.hillas.intensity > 500"),
+        ],
+        help=(
+            "list of tuples of ('<description', 'expression string') to accept "
+            "(select) a given data value.  E.g. `[('mycut', 'x > 3'),]. "
+            "You may use `numpy` as `np` and `astropy.units` as `u`, but no other"
+            " modules."
+        ),
+    ).tag(config=True)
+
+
+class RingQuery(QualityQuery):
+    quality_criteria = List(
+        Tuple(Unicode(), Unicode()),
+        default_value=[
+            ("radius_not_nan", "np.isfinite(ring.radius.value)"),
+            ("min_pixels", "np.count_nonzero(mask) > 50"),
+            ("ring_containment", "parameters.containment > 0.5"),
+        ],
+        help=(
+            "list of tuples of ('<description', 'expression string') to accept "
+            "(select) a given data value.  E.g. `[('mycut', 'x > 3'),]. "
+            "You may use `numpy` as `np` and `astropy.units` as `u`, but no other"
+            " modules."
+        ),
+    ).tag(config=True)
+
+
+class MuonProcessor(TelescopeComponent):
+
+    completeness_threshold = FloatTelescopeParameter(
+        default_value=30.0, help="Threshold for calculating the ``ring_completeness``"
+    ).tag(config=True)
+
+    ratio_width = FloatTelescopeParameter(
+        default_value=1.5,
+        help=(
+            "Ring width for intensity ratio"
+            " computation as multiple of pixel diameter"
+        ),
+    ).tag(config=True)
+
+    pedestal = FloatTelescopeParameter(
+        help="Pedestal noise rms", default_value=1.1
+    ).tag(config=True)
+
+    def __init__(self, subarray, **kwargs):
+        super().__init__(subarray, **kwargs)
+        self.dl1_query = DL1Query(parent=self)
+        self.ring_query = RingQuery(parent=self)
+        self.geometries = {
+            tel_id: tel.camera.geometry.transform_to(TelescopeFrame())
+            for tel_id, tel in self.subarray.tel.items()
+        }
+        self.fov_radius = {
+            tel_id: geometry.guess_radius()
+            for tel_id, geometry in self.geometries.items()
+        }
+
+        self.ring_fitter = MuonRingFitter(parent=self)
+
+        self.intensity_fitter = MuonIntensityFitter(subarray=subarray, parent=self)
+
+    def __call__(self, event: ArrayEventContainer):
+        for tel_id in event.dl1.tel:
+            self._process_telescope_event(event, tel_id)
+
+    def _process_telescope_event(self, event, tel_id):
+        event_index = event.index
+        event_id = event_index.event_id
+
+        if self.subarray.tel[tel_id].optics.n_mirrors != 1:
+            self.log.warn(
+                f"Skipping non-single mirror telescope {tel_id}"
+                " set --allowed_tels to get rid of this warning"
```

Comment:
This sounds as if setting allowed tels to include dual mirror telescopes would then process dual mirror telescopes.  I assume you want to say to set allowed tels to *only* include single mirror telescopes?

Reply:
Maybe: `"Exclude dual mirror telescopes via setting `--allowed-tels`?"

Reply:
also, would be good to say something that explains why this is, for example:  "muon-fitting for dual-mirror telescope is currently not supported. "   That way it is clear that it *will be supported in the future* (it's simply not yet implemented).